src/components/Modal/__stories__/modal.stories.mdx

```diff
@@ -0,0 +1,308 @@
+
+<Canvas>
+  <Story name="Overview">
+    {() => {
+      const { openModalButton, modalButtons, show, closeModal, ref } = useModalHelper();
+      return (
+        <div>
+          {openModalButton}
+          <Modal
+            id={"story-book-modal"}
+            title={"Modal title"}
+            triggerElement={ref.current}
+            show={show}
+            onClose={closeModal}
+            closeButtonAriaLabel={"close"}
+          >
+            <ModalContent>Modal content goes here</ModalContent>
+            <ModalFooter>{modalButtons}</ModalFooter>
+          </Modal>
+        </div>
+      );
+    }}
+  </Story>
+</Canvas>
+
+## Props
+
+<ArgsTable of={Modal} />
+
+## Usage
+
+### Showing the Modal
+
+You show/hide the Modal by passing the <b>show</b> prop. This gives you complete control on when to show and hide the Modal.
+One thing to consider is when the user initiate closing the Modal by
+
+### Modal content
+
+### Design and customizations
+
+<UsageGuidelines guidelines={["Usage guideline 1", "Usage guideline 2", "Usage guideline 3"]} />
+
+<Tip title="Not what you were looking for?">Tip content</Tip>
+
+## Variants
+
+### Alert Modal (Modal)
+
+This Modal will NOT be closed when you click outside or click ESC key. Only way to close it is to use the close button.
+
+<UsageGuidelines guidelines={["set `isAlertDialog` to `true` in order to turn a Modal to modal."]} />
+
+<Canvas>
+  <Story name="Alert Modal">
+    {() => {
+      const { openModalButton, modalButtons, show, closeModal, ref } = useModalHelper();
+      return (
+        <div>
+          {openModalButton}
+          <Modal
+            id={"story-book-modal"}
+            title={"Modal title"}
+            triggerElement={ref.current}
+            show={show}
+            onClose={closeModal}
+            isAlertDialog
+            closeButtonAriaLabel={"close"}
+          >
+            <p>Use the close button to dismiss this Modal.</p>
+            <ModalFooter>{modalButtons}</ModalFooter>
+          </Modal>
+        </div>
+      );
+    }}
+  </Story>
+</Canvas>
+
+### Modal with a form
+
+<UsageGuidelines
+  guidelines={[
+    "The Modal is capturing the focus. Use TAB ket to jump throw the close button and the inputs.",
+    "Use Alert Modal so the user won't close the Modal by mistake and loose changes."
+  ]}
+/>
+
+<Canvas>
+  <Story name="Form Modal">
+    {() => {
+      const { openModalButton, modalButtons, show, closeModal, ref } = useModalHelper();
+      return (
+        <div>
+          {openModalButton}
+          <Modal
+            id={"story-book-modal"}
+            title={"Modal title"}
+            triggerElement={ref.current}
+            show={show}
+            onClose={closeModal}
+            isAlertDialog
+            closeButtonAriaLabel={"close"}
+          >
+            <ModalContent>
+              <form>
+                <TextField title="Your Name" autoFocus size={TextField.sizes.MEDIUM} placeholder="John Doe" />
```

Comment:
I would duplicate `inputAriaLabel` from `title` or from `title` + `placeholder`.
In fact I've seen that in TextField `ariaLabel = inputAriaLabel || placeholder`.  I think it's a strange implementation, cause voice over will always pronounce placeholder even when it was cleared and never pronounce the title, which might give more understanding. Maybe `title` should be somewhere there too, what do you think? Smth like  `ariaLabel = inputAriaLabel || title` or `ariaLabel = inputAriaLabel || "${titile}, ${placeholder}"`?